rest/client/request.go

```diff
@@ -56,6 +57,7 @@ func (c *communicatorImpl) newRequest(method, path, taskSecret, version string,
 		if err != nil {
 			return nil, err
 		}
+		r.Header.Add(evergreen.ContentLengthHeader, strconv.Itoa(len(out)))
```

Comment:
i don't think this is needed, see other comment

Reply:
The population of r.ContentLength is set during Request, and the body isn't reliably set here (in this case,) so I think it is needed.

Reply:
done

Reply:
Ah i see, since NewRequest is called with nil data it doesn't set it by default. Personally i'd reorder things so NewRequest is called with a byte string, but it's not a big deal.